vividus-extension-selenium/src/main/java/org/vividus/selenium/manager/AbstractWebDriverManagerContext.java

```diff
@@ -0,0 +1,60 @@
+package org.vividus.selenium.manager;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.vividus.testcontext.TestContext;
+
+public abstract class AbstractWebDriverManagerContext<K extends HasContextKey> implements WebDriverManagerContext<K>
+{
+    private final TestContext testContext;
+    private final Class<?> storageType;
+
+    public AbstractWebDriverManagerContext(TestContext testContext, Class<?> storageType)
+    {
+        this.testContext = testContext;
+        this.storageType = storageType;
+    }
+
+    @Override
+    public <T> T get(K key)
+    {
+        return testContext.get(key.getContextKey());
+    }
+
+    @Override
+    public <T> void put(K key, T value)
+    {
+        testContext.put(key.getContextKey(), value);
+        getKeys().add(key);
+    }
+
+    public <T> T get(K key, Supplier<T> initialValueSupplier)
+    {
+        getKeys().add(key);
+        return testContext.get(key.getContextKey(), initialValueSupplier);
+    }
+
+    @Override
+    public void reset()
+    {
+        getKeys().stream().map(HasContextKey::getContextKey).forEach(testContext::remove);
+    }
+
+    @Override
+    public void reset(K key)
+    {
+        testContext.remove(key.getContextKey());
+    }
+
+    protected TestContext getTestContext()
+    {
+        return testContext;
+    }
+
+    private Set<K> getKeys()
+    {
+        return testContext.get(storageType, HashSet::new);
+    }
```

Comment:
what?

Reply:
Updated to use SimpleTestContext instead of Set